src/main/java/com/example/ForexApp_v1/service/TransacService.java

```diff
@@ -27,13 +29,16 @@ public TransacService(TransacRepository transacRepository, CurrencyDownloadApi c
         this.jsonMapper = jsonMapper;
     }
 
-    public Transac addTransaction(TransacDTO transacDTO){
+    public TransacDTO addTransaction(TransacDTO transacDTO){
         Transac transac = calculateTransaction.manualCalculate(transacDTO.getValueCurrency(), transacDTO.getCodeCurrency(), (transacDTO.getDateTransaction().toString()).substring(0,10));
         transacRepository.createOrUpdate(transac);
-        return transac;
+
+        TransacDTO result = new TransacDTO(transac.getDateTransaction().toString(), transac.getCodeCurrency(), transac.getValueCurrency(), transac.getId(), transac.getValuePln(), transac.getResultTransaction(), transac.getIsDone());
```

Comment:
you could just return the value of new TransacDTO, no need for a variable in the meantime

Reply:
you're right! thanks :)